### PR TITLE
[Dataset manager] Fix import for MOTS format

### LINKED ISSUE #3360: AssertionError when dumping a dataset after I uploaded a MOTS annotation.
### My actions before raising this issue
- [x] Read/searched [the docs](https://github.com/opencv/cvat/tree/master#documentation)
- [x] Searched [past issues](/issues)

### Expected Behaviour

Dump to COCO 1.0 JSON

### Current Behaviour

image
This assertion error shows up, no matter what kind of dump I choose. The error that shows up before that is `assert shape["frame"] > curr_frame`.
I have 8 other datasets that have worked fine, with no errors whatsoever. Only this one has this problem for some reason.

### Steps to Reproduce (for bugs)

1. Download the images and annotations (https://drive.google.com/drive/folders/1ujcG858m4xe8COTaf1PxONyC78mV0a7N?usp=sharing)
2. add apple to the labels
3. Upload the images
4. Upload the annotation zip (instances.zip) by going to upload annotations -> MOTS PNG 1.0
5. Dump annotations -> COCO 1.0. Then failing with an assertion error.

### Context

I want to get the COCO annotations from an annotated MOTS PNG dataset.

### Your Environment

I tried it on two environments with the same outcome:
1. Directly in https://cvat.org/
2. In my Linux computer, running Ubuntu 18.04, the latest Docker (version 20).

### PR DESCRIPTION
**Pull request by kirill-sizov**
### Motivation and context
Fixed #3360.
Some masks in the MOTS PNG format may contain several polygons with the same track_id, for such annotations to be loaded into CVAT, they must be grouped during export. This is what I did in this PR.

### Checklist

- [ ] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [x] I have added tests to cover my changes
- [x] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- ~[ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))~

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Review by zhiltsov-max — APPROVED** [review]
Looks working, but I think, it deserves a test.

**Comment by kirill-sizov** [comment]
> Looks working, but I think, it deserves a test.

Yes is it, but current CVAT tests check if CVAT can upload annotations dumped by CVAT.
Should we run tests for uploading different various of external datasets in supported formats?

File: cvat/apps/dataset_manager/tests/test_formats.py [hunk 1]
```diff
@@ -598,9 +599,10 @@ def _generate_custom_annotations(self, annotations, task):
         self._put_api_v1_task_id_annotations(task["id"], annotations)
         return annotations
 
-    def _generate_task_images(self, count, name="image"):
+    def _generate_task_images(self, count, name="image", **kwargs):
```
Comment (zhiltsov-max) [inline cvat/apps/dataset_manager/tests/test_formats.py:602]: I suggest to rename `kwargs` to `image_params` or so.

File: cvat/apps/dataset_manager/tests/test_formats.py [hunk 2]
```diff
@@ -912,7 +914,44 @@ def test_can_import_annotations_for_image_with_dots_in_filename(self):
...
+                    'Kitti Raw Format 1.0'}:
                     self.skipTest("Format is disabled")
 
                 self._test_can_import_annotations(task, format_name)
+
+    def test_can_import_mots_annotations_with_splited_masks(self):
+        #https://github.com/openvinotoolkit/cvat/issues/3360
+        from datumaro.components.dataset import Dataset, DatasetItem
+        from datumaro.components.extractor import Mask
+        from numpy import array
```
Comment (zhiltsov-max) [inline cvat/apps/dataset_manager/tests/test_formats.py:927]: What is the point of such import?

File: cvat/apps/dataset_manager/tests/test_formats.py [hunk 3]
```diff
@@ -912,7 +914,44 @@ def test_can_import_annotations_for_image_with_dots_in_filename(self):
             self._generate_annotations(task, format_name)
 
             with self.subTest(format=format_name):
-                if not f.ENABLED:
+                if not f.ENABLED or format_name in {'Sly Point Cloud Format 1.0',
+                    'Kitti Raw Format 1.0'}:
```
[the remark this answers is not in the document]
Comment (kirill-sizov) [reply cvat/apps/dataset_manager/tests/test_formats.py:918]: This test was wrong. After I fixed it it turned out that these two formats fail. I'll investigate this issue, but it's not related with this PR.

File: cvat/apps/dataset_manager/tests/test_formats.py (hunk 2 above)
Comment (kirill-sizov) [reply cvat/apps/dataset_manager/tests/test_formats.py:927]: These classes are only needed in this test, so I decided to import them inside the function.

Comment (zhiltsov-max) [reply cvat/apps/dataset_manager/tests/test_formats.py:926]: It's not a significant reason for having them here.

Also, numpy is typically imported as `import numpy as np`

File: cvat/apps/dataset_manager/bindings.py [hunk 4]
```diff
@@ -908,7 +908,7 @@ def _make_image(i, **kwargs):
                         attributes["labels"].append({"label_id": idx, "name": label["name"], "color": label["color"]})
                         attributes["track_id"] = -1
 
-                dm_item = datumaro.DatasetItem(id=osp.split(frame_data.name)[-1].split('.')[0],
+                dm_item = datumaro.DatasetItem(id=osp.splitext(osp.split(frame_data.name)[-1])[0],
```
Comment (zhiltsov-max) [inline cvat/apps/dataset_manager/bindings.py:911]: Probably, this should not be in this PR.

Comment (kirill-sizov) [reply cvat/apps/dataset_manager/bindings.py:911]: Okay, I'll made these changes in new PR

**Review by zhiltsov-max — APPROVED** [review]

**Pull request merged**